### ISSUE TITLE
Error in eigen(corr) : infinite or missing values in 'x'

### ISSUE THREAD

**Issue by sunxi822** [taiyun/corrplot#182]
Hi taiyun,

I am using **corrplot** to visualizing correlations among different traits. 

My raw data has NA values, but it could work with corrplot as follows:

`corrplot(contrib$paircor,order = "AOE",type="upper",tl.pos = "d")`
`corrplot(contrib$paircor,add=TRUE, type="lower", method="number",order="AOE",diag=FALSE,tl.pos="n", cl.pos="n",tl.cex=10,tl.col = "black")`

However, after the log10 transformation of my raw data, when I ran the code, I got the following errors:
`Error in eigen(corr) : infinite or missing values in 'x'`

I have tried to solve this problem by changing my code as follows:
`corrplot(cor(contrib$paircor,use = "complete.obs"),order = "AOE",type="upper",tl.pos = "d")
corrplot(cor(contrib$paircor,use = "complete.obs"),add=TRUE, type="lower", method="number",order="AOE",diag=FALSE,tl.pos="n", cl.pos="n",tl.cex=10,tl.col = "black")`
but I failed and got a error.

I don't know why this error occurred, Hope for your reply.

Thanks!

**Comment by taiyun** [comment]
Can you give a repeatable example with data and code？

**Comment by sunxi822** [comment]
Thanks for your reply!

Before upload the data and code,   I checked my data. 

And I found that one data should be `NA` was written as `0` mistakenly. This caused  `-inf` in the log10 transformation data.  

If I use data including `-inf` to calculate the correlations between traits,` NA` will appear in the correlation matrix.  

`NA` in the correlation matrix leads to  `Error in eigen(corr) : infinite or missing values in 'x' `.

I corrected the data and the problem was solved.

Many thanks!
